src/lib/structures/managers/StickyRoleManager.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @typescript-eslint/explicit-member-accessibility */
+import { GuildSettings, StickyRole } from '@lib/types/settings/GuildSettings';
+import { Guild } from 'discord.js';
+
+export class StickyRoleManager {
+
+	#guild: Guild;
+
+	public constructor(guild: Guild) {
+		this.#guild = guild;
+	}
+
+	private get entries() {
+		return this.#guild.settings.get(GuildSettings.StickyRoles);
+	}
+
+	public get(userID: string): readonly string[] {
+		return this.entries.find(entry => entry.user === userID)?.roles ?? [];
+	}
+
+	public async fetch(userID: string, extraContext?: unknown): Promise<readonly string[]> {
+		const arrayIndex = this.getKey(userID);
+		if (arrayIndex === -1) return [];
+
+		const entry = this.entries[arrayIndex];
+		const roles = [...this.cleanRoles(entry.roles)];
+		if (entry.roles.length === roles.length) return entry.roles;
+
+		const clone: StickyRole = { user: userID, roles };
+		await this.#guild.settings.update(GuildSettings.StickyRoles, clone, { arrayIndex, extraContext });
```

Comment:
For consistency please wrap this over multiple lines like in other places where update is used